src/index.js

```diff
@@ -1,122 +1,142 @@
+ const divInput = document.querySelector(`.inputBox${chores.index}`)
+ const trashCan = document.querySelector(`.del-btn-${chores.index}`);
+
+
+
+
+const checkbox = document.querySelector(`#check${chores.index}`);
+if(checkbox.textContent === '\u2714'){
+    checkbox.nextElementSibling.classList.add('crossinput');
+}
+checkbox.addEventListener('click', function(e){
+//    const btnindex = e.target.id.slice(5);
+   editBtn.classList.toggle('edit-display');
+   trashCan.classList.toggle('del-display');
+
+   if(e.target.textContent === '\u2714'){
+       
+       e.target.textContent= ' ';
+       divInput.classList.remove('crossinput');
+       chores.completed = false;
+       const localData = JSON.parse(localStorage.getItem('listItem'));
+       localData[chores.index -1].completed = false;
+       localStorage.setItem('listItem', JSON.stringify(localData));
+     
+       console.log(chores.completed)
+       
+   }else{
+    e.target.textContent= '\u2714';
+    divInput.classList.add('crossinput');
+    chores.completed = true;
+    const localData = JSON.parse(localStorage.getItem('listItem'));
+    localData[chores.index - 1].completed = true;
+    localStorage.setItem('listItem', JSON.stringify(localData))
+
+    console.log(chores.completed)
+   }
+})
+editBtn.addEventListener('click',()=>{
+    if(divInput.hasAttribute('readonly', true)){
+        divInput.parentElement.parentElement.classList.add('activebg');
+        divInput.removeAttribute('readonly');
+        divInput.addEventListener('keypress', (e)=>{
+
+                 if(e.key === 'Enter'){
+                    console.log("attribute removed")
+                    divInput.removeAttribute('readonly');
+                    divInput.parentElement.parentElement.classList.remove('activebg');
+
+                    const localData = JSON.parse(localStorage.getItem('listItem'));
+                    console.log(divInput.value, chores.index)
+                    localData[chores.index - 1].description = divInput.value;
+                    localStorage.setItem('listItem', JSON.stringify(localData))
+                    divInput.removeAttribute('readonly');
+                
+                            
+            }
+        })
+        console.log("yea")
     }
     else{
-        e.target.innerHTML = "edit"
+        divInput.setAttribute('readonly', true)
+        console.log("attribute removed");
 
     }
-}
+   
+        
+} )
+
+// -------------------------clear btn-----------------
+
+clrBtn.addEventListener('click', ()=>{
+
+    console.log("clear botton clicked")
+    if(checkbox.textContent === '\u2714'){
+        checkbox.parentNode.parentNode.remove();
+
+    }
+    const localData = JSON.parse(localStorage.getItem('listItem'));
+    const deletedLocalData = localData.filter((item) => {
+        if (item.completed) {
+          return null;
+        }
+        return item;
+      });
+      localStorage.setItem('listItem', JSON.stringify(deletedLocalData))
+
+    
```

Comment:
- [ ] (Optional) It would be nice to reset the `index` key of the remaining tasks in the array after removing all completed tasks. You can do that by using a `forEach` which takes an `index` as a second parameter to loop through the `deletedLocalData` array and setting each task's `index` key to the index position of the item + 1. 👍